src/coreclr/vm/commtmemberinfomap.cpp

```diff
@@ -607,6 +620,15 @@ void ComMTMemberInfoMap::SetupPropsForInterface(size_t sizeOfPtr)
     for (iMD=0; iMD < nSlots; ++iMD)
     {
         pMeth = m_MethodProps[iMD].pMeth;
+        if (pMeth == nullptr)
+        {
+            // For async methods, we skip .NET methods when building the COM vtable.
+            // So at some point we hit the end of the .NET methods before we run
+            // through all possible vtable slots.
+            // Record when we ran out here.
+            nSlots = iMD + 1;
```

Comment:
Do you think it is worth looping through the remainder of the list and verify all are `nullptr` under a `_DEBUG` check?